jerry-core/jerry.c

```diff
@@ -2087,3 +2087,46 @@ jerry_dispatch_object_free_callback (ecma_external_pointer_t freecb_p, /**< poin
 
   jerry_make_api_available ();
 } /* jerry_dispatch_object_free_callback */
+
+/**
+ * Reserve a slot in the context for a module
+ *
+ * Returns: 0 or positive on success, -1 on failure.
+ */
+
+/* This value is context-independent */
+static int module_index = -1;
```

Comment:
Because the index counts the modules that were registered in the process not the instances of the modules that were started in a context.

I originally had the index in the context but I realized that `jerry_register_module()` will only be called once for each module, no matter how many contexts are later created.